<launch>

  <param name="/use_sim_time" value="false"/>
  
  <arg name="mpc_frequency" value="10" />
  <arg name="pre_step"      value="20" />
  <arg name="step_time"     value="0.2" />
  <!-- can choose None 0, MPC-DC 1, MPC-SCBF 2, MPC-DCBF 3, MPC-ACBF 4 -->
  <arg name="controller"    value="3" />
  <arg name="front_dis"     value="false" />
  <arg name="front_adsm"    value="false" />
  <arg name="static_path"   value="false" />


  <!-- MPC-CBF C++版本 -->
  <node name="local_planner" pkg="mpc_dcbf" type="mpc_node_c" output="screen">
    <param name="mpc/controller_type" value="3"/>
    <param name="mpc/mpc_frequency" value="$(arg mpc_frequency)" />
    <param name="mpc/pre_step"    value="$(arg pre_step)" />
    <param name="mpc/step_time"   value="$(arg step_time)" />
    <param name="mpc/gamma"       value="0.25" />
    <param name="mpc/tau_scale"   value="0.0" />
  
    <remap from="/Odometry" to="/Odometry" />
    <remap from="/obs_Manager_node/obs_predict_pub" to="/globalFsm_by_adsm/obs_predict_pub" />
    <remap from="/local_path" to="/local_path" />
    <remap from="/cmd_vel"    to="/cmd_vel1" />
  </node>

  <!-- <node pkg="rviz" type="rviz" name="rviz" output="log" args="-d $(find swarm_test)/config/acbf_plan.rviz" /> -->

</launch>
